sdk/storage/Azure.Storage.Webjobs.Extensions/src/Microsoft.Azure.WebJobs.Extensions.Storage/Blobs/Bindings/StorageBlobContainerExtensions.cs

```diff
@@ -35,24 +37,38 @@ internal static class StorageBlobContainerExtensions
             }
         }
 
-        private static async Task<ICloudBlob> GetExistingOrNewBlockBlobReferenceAsync(CloudBlobContainer container,
+        public static async Task<(BlobBaseClient, BlobProperties)> GetBlobReferenceFromServerAsync(this BlobContainerClient container, string blobName, CancellationToken cancellationToken = default)
```

Comment:
This is cool, what's the name of the concept of returning two objects from a method?

Reply:
Tuple: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/value-tuples .
However, I was trying to not abuse this. I.e. I had more of this at some point elsewhere (different pair  - container and blob) so I created named class for that (especially when this flows between many classes).
This one is left because usage is quite localized and easy to follow (for now).